docs/source/cost_functions.rst

```diff
@@ -0,0 +1,80 @@
+.. Copyright 2018 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+==============
+Cost functions
+==============
+
+Relative Transform Error 2D
+===========================
+
+Given two poses
+:math:`\mathbf{p_i} = [\mathbf{x_i}; \theta_i] = [x_i, y_i, \theta_i]^T`
+and :math:`\mathbf{p_j} = [\mathbf{x_j}; \theta_j] = [x_j, y_j, \theta_j]^T`
+the transformation :math:`\mathbf T` from the coordinate frame :math:`j` to the
+coordinate frame :math:`i` has the following form
+
+.. math::
+ \mathbf{T}( \mathbf{p_i},\mathbf{p_j}) =
+ \left[
+   \begin{array}{c}
+        R(\theta_i)^T (\mathbf x_j - \mathbf x_i) \\
+        \theta_j-\theta_i
+   \end{array}
+ \right]
+where :math:`R(\theta_i)^T` is the rotation matrix of :math:`\theta_i`.
+
+The weighted error :math:`f:\mathbb R^6 \mapsto \mathbb R^3` between
```

Comment:
Maybe worth mentioning the order of the elements in the 6-vector [x_i, y_i \theta_i, x_j, y_j, \theta_j]^T to ease matching to the Jacobian entries below.